cmake/Modules/FindMagic_EP.cmake

```diff
@@ -35,12 +35,39 @@
 include(TileDBCommon)
 
 if(TILEDB_VCPKG)
-  find_package(unofficial-libmagic CONFIG REQUIRED)
-  set(libmagic_DICTIONARY ${unofficial-libmagic_DICTIONARY})
+  find_package(unofficial-libmagic CONFIG)
```

Comment:
Should we gate this behind a check that we're building on mingw? I don't think its a huge issue either way though.

Reply:
I prefer not having the build script know about MinGW because it provides flexibility. If the easy to use `unofficial-libmagic` package exists, we can use it, otherwise we fall back to the traditional searching of the library.

Reply:
I hate to make so much work for the R support. I should add that we still use a dated ("Rtools40" for R 4.0.*) builder by inertia, the current one is Rtools43 and there is more pacman and whatnot support.

Maybe we are better off doing a minimal "hack" here given that mingw is a bit of a one-off, and then maybe try to update the builder?

Reply:
@teo-tsirpanis Fair point. That's more than good enough for me to leave it as is.